torch/distributed/rpc/api.py

```diff
@@ -207,6 +208,7 @@ def shutdown(graceful=True):
 
     if graceful:
         _wait_all_workers()
+        _delete_all_user_rrefs()
         _agent.join()
```

Comment:
If `ThriftAgent.join()` is no-op, how will this work for that?

Reply:
This PR will not help ThriftAgent unfortunately, as I have tracked this situation in https://github.com/pytorch/pytorch/issues/31325#issuecomment-574918917.
I must update `_wait_all_workers()` as the next step.